src/middlewares/createInsightsMiddleware.ts

```diff
@@ -76,28 +85,19 @@ aa('setUserToken', 'your-user-token');
           .getHelper()!
           .setQueryParameter('clickAnalytics', true);
 
-        if (hasInsightsClient) {
+        const anonymousUserToken = getInsightsAnonymousUserTokenInternal();
+        if (hasInsightsClient && anonymousUserToken) {
           // When `aa('init', { ... })` is called, it creates an anonymous user token in cookie.
           // We can set it as userToken.
-          setUserTokenToSearch(getInsightsAnonymousUserTokenInternal());
+          setUserTokenToSearch(anonymousUserToken);
+        }
+
+        if (queuedUserToken) {
```

Comment:
Why can this condition and the one below be satisfied at the same time? It makes the flow harder to understand.

If that what it means, I'd suggest changing to this:

```ts
// We consider the `userToken` coming from a `init` call to have a higher
// importance than the one coming from the queue.
if (userTokenBeforeInit) {
  insightsClient('setUserToken', userTokenBeforeInit);
} else if (queuedUserToken) {
  insightsClient('setUserToken', queuedUserToken);
}
```

Reply:
https://github.com/algolia/instantsearch.js/pull/4575/commits/8b10f8940ac1b6e94ab8cb3757688207f57353fd